earlydecoder/ephemeral_resource.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package earlydecoder
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-schema/module"
+)
+
+type ephemeralResource struct {
+	Type     string
+	Name     string
+	Provider module.ProviderRef
+}
+
+// MapKey returns a string that can be used to uniquely identify the receiver
+// in a map[string]*ephemeralResource.
+func (r *ephemeralResource) MapKey() string {
+	return fmt.Sprintf("%s.%s", r.Type, r.Name)
```

Comment:
```suggestion
	return fmt.Sprintf("ephemeral.%s.%s", r.Type, r.Name)
```

While the current string shouldn't cause any collisions, I think we should still add the `ephemeral` prefix

Reply:
Oh, yeah, that was a copy&paste oversight 😄 Fixed!